netlogo-gui/src/main/app/Tabs.scala

```diff
@@ -235,39 +262,55 @@ class Tabs(val workspace:    GUIWorkspace,
     menu.offerAction(newAction)
   }
 
-  private def stripPath(filename: String): String =
-    filename.substring(filename.lastIndexOf(System.getProperty("file.separator")) + 1, filename.length)
-
-  val printAction = {
-    val action = RichAction("print-current-tab") { _ =>
-      currentTab match {
-        case printable: org.nlogo.swing.Printable =>
-          try org.nlogo.swing.PrinterManager.print(printable, workspace.modelNameForDisplay)
-          catch {
-            case abortEx: java.awt.print.PrinterAbortException => org.nlogo.api.Exceptions.ignore(abortEx)
-          }
-      }
-    }
-
-    action.putValue(Action.NAME,            I18N.gui.get("menu.file.print"))
-    action.putValue(ActionCategoryKey,      UserAction.FileCategory)
-    action.putValue(ActionGroupKey,         "org.nlogo.app.Tabs.Print")
-    action.putValue(Action.ACCELERATOR_KEY, UserAction.KeyBindings.keystroke('P', withMenu = true))
-    action
-  }
-
-  override def processMouseMotionEvent(e: java.awt.event.MouseEvent) {
+  override def processMouseMotionEvent(e: MouseEvent) {
     // do nothing.  mouse moves are for some reason causing doLayout to be called in the tabbed
     // components on windows and linux (but not Mac) in java 6 it never did this before and I don't
     // see any reason why it needs to. It's causing flickering in the info tabs on the affected
     // platforms ev 2/2/09
   }
 
-  def handle(e: ModelSavedEvent) {
-    saveExternalFiles()
-  }
-
   def handle(e: AfterLoadEvent) {
     requestFocus()
   }
+
+  private val saveModelActions = {
+    def saveAction(saveAs: Boolean) =
+      new ExceptionCatchingAction(if (saveAs) I18N.gui.get("menu.file.saveAs") + Ellipsis else I18N.gui.get("menu.file.save"), Tabs.this)
```

Comment:
Can this class be defined in `FileManager` instead of `Tabs`? `Tabs` is already big enough and this is not central to what `Tabs` does. It can be instantiated in here passing in `this` / `Tabs.this` as a constructor parameter.

This line needs at least one line break, maybe 2. Lines *should* be 80 characters or less where possible and should *never* be longer than 120 characters except in extreme circumstances.